packages/@uppy/core/src/index.js

```diff
@@ -1708,6 +1701,23 @@ class Uppy {
   }
 }
 
+/**
+ * Back compat for when uppy.state is used instead of uppy.getState().
+ */
+Reflect.defineProperty(Uppy.prototype, 'state', {
```

Comment:
So why did you go for `Reflect.defineProperty`? Most of these things like Symbol and Relfect are new to me, might be for other contributors as well, maybe we should add a bit of comments around them?

Reply:
I thought `state` was enumerable, and using `Reflect.defineProperty` was the way to make it non-enumerable. After checking, it turns out it was already non-enumerable 😳 apparently class getters are different from object getters (ECMAScript is full of surprises): 
```js
Reflect.getOwnPropertyDescriptor(class{ get state(){} }.prototype, 'state');
// {
//    configurable: true,
//    enumerable: false,
//    get: [function "get state"],
//    set: undefined,
// }
Reflect.getOwnPropertyDescriptor({ get state(){} }, 'state');
// {
//    configurable: true,
//    enumerable: true,
//    get: [function "get state"],
//    set: undefined,
// }
```

Reasons to not use `Reflect.defineProperty`:
- we don't need it to make the property non-enumerable
-  it makes the code harder to read for someone not familiar with this API.

Reasons to use `Reflect.defineProperty`:
- it makes more obvious that `state` is not enumerable.
- it allows to reuse `getState` instead of re-defining a getter – I would expect it to marginally reduce the memory footprint.

wdyt?